### DISCUSSION TITLE
Conditional layers not working in podman

### DISCUSSION THREAD

**Discussion by dominikjeske** [containers/podman#17732] [Q&A]
I have Dockerfile that is working on Windows (Docker Desktop) but is failing when using Podman on Linux (Redhat with podman 4.2.0)

```
ARG USE_NUGET_REFERENCES=false

FROM baseimage AS base

#...

FROM base as build

#...

# (Conditional) Use project references
FROM build AS build-nugets-false

#...

FROM build AS build-nugets-true

#...

# Basing on USE_NUGET_REFERENCES we decide what mode we are using
FROM build-nugets-${USE_NUGET_REFERENCES} AS build-final
```

This Docker file is using condition based on input argument. On Windows it is working and with podman it tries to get build-nugets-false from repository instead using local layer. Is it bug or by design?

`[5/7] STEP 1/20: FROM build-nugets-false AS build-final
Resolving "build-nugets-false" using unqualified-search registries (/etc/containers/registries.conf)
Trying to pull registry.access.redhat.com/build-nugets-false:latest...
time="2023-02-28T15:44:11+01:00" level=warning msg="Failed, retrying in 2s ... (1/3). Error: initializing source docker://registry.access.redhat.com/build-nugets-false:latest: pinging container registry registry.access.redhat.com: Get \"https://registry.access.redhat.com/v2/\": dial tcp 95.100.111.121:443: i/o timeout"
time="2023-02-28T15:45:13+01:00" level=warning msg="Failed, retrying in 2s ... (2/3). Error: initializing source docker://registry.access.redhat.com/build-nugets-false:latest: pinging container registry registry.access.redhat.com: Get \"https://registry.access.redhat.com/v2/\": dial tcp 95.100.111.99:443: i/o timeout"
time="2023-02-28T15:46:15+01:00" level=warning msg="Failed, retrying in 2s ... (3/3). Error: initializing source docker://registry.access.redhat.com/build-nugets-false:latest: pinging container registry registry.access.redhat.com: Get \"https://registry.access.redhat.com/v2/\": dial tcp 95.100.111.121:443: i/o timeout"
Trying to pull registry.redhat.io/build-nugets-false:latest...
time="2023-02-28T15:48:17+01:00" level=warning msg="Failed, retrying in 2s ... (1/3). Error: initializing source docker://registry.redhat.io/build-nugets-false:latest: pinging container registry registry.redhat.io: Get \"https://registry.redhat.io/v2/\": dial tcp 23.200.161.167:443: i/o timeout"
time="2023-02-28T15:49:19+01:00" level=warning msg="Failed, retrying in 2s ... (2/3). Error: initializing source docker://registry.redhat.io/build-nugets-false:latest: pinging container registry registry.redhat.io: Get \"https://registry.redhat.io/v2/\": dial tcp 23.200.161.167:443: i/o timeout"
time="2023-02-28T15:50:21+01:00" level=warning msg="Failed, retrying in 2s ... (3/3). Error: initializing source docker://registry.redhat.io/build-nugets-false:latest: pinging container registry registry.redhat.io: Get \"https://registry.redhat.io/v2/\": dial tcp 23.200.161.167:443: i/o timeout"
Trying to pull docker.io/library/build-nugets-false:latest...
time="2023-02-28T15:52:23+01:00" level=warning msg="Failed, retrying in 2s ... (1/3). Error: initializing source docker://build-nugets-false:latest: pinging container registry registry-1.docker.io: Get \"https://registry-1.docker.io/v2/\": dial tcp 34.205.13.154:443: i/o timeout"
time="2023-02-28T15:53:25+01:00" level=warning msg="Failed, retrying in 2s ... (2/3). Error: initializing source docker://build-nugets-false:latest: pinging container registry registry-1.docker.io: Get \"https://registry-1.docker.io/v2/\": dial tcp 3.216.34.172:443: i/o timeout"
time="2023-02-28T15:54:27+01:00" level=warning msg="Failed, retrying in 2s ... (3/3). Error: initializing source docker://build-nugets-false:latest: pinging container registry registry-1.docker.io: Get \"https://registry-1.docker.io/v2/\": dial tcp 34.205.13.154:443: i/o timeout"
Error: error creating build container: 3 errors occurred while pulling:
 * initializing source docker://registry.access.redhat.com/build-nugets-false:latest: pinging container registry registry.access.redhat.com: Get "https://registry.access.redhat.com/v2/": dial tcp 95.100.111.121:443: i/o timeout
 * initializing source docker://registry.redhat.io/build-nugets-false:latest: pinging container registry registry.redhat.io: Get "https://registry.redhat.io/v2/": dial tcp 23.200.161.167:443: i/o timeout
 * initializing source docker://build-nugets-false:latest: pinging container registry registry-1.docker.io: Get "https://registry-1.docker.io/v2/": dial tcp 34.205.13.154:443: i/o timeout
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.`

**Comment by rhatdan** [comment]
@flouthoc PTAL
